test/getCommands.spec.js

```diff
@@ -1,50 +1,46 @@
 const path = require('path');
-const Module = require('module');
 const expect = require('chai').expect;
 const getCommands = require('../src/getCommands');
-const findPlugins = require('../src/findPlugins');
-const mockRequire = require('mock-require');
-const mockFs = require('mock-fs');
+const mock = require('mock-require');
 
-const commands = getCommands();
-const filesPath = path.join(__dirname, 'fixtures', 'files');
-const testPlugin = findPlugins(filesPath);
-const testPluginPath = path.join(__dirname, '..', 'node_modules', 'rnpm-plugin-test');
-const singlePlugin = require(path.join(filesPath, 'plugin'));
-const multiPlugin = require(path.join(filesPath, 'plugins'));
+const commands = require('./fixtures/commands');
+const plugins = require('./fixtures/plugins');
 
-describe('getCommands', () => {
+const pluginFolder = path.join(process.cwd(), 'node_modules', 'rnpm-plugin-link');
 
-  before(() => {
-    mockRequire(path.join(__dirname, '..'), {});
-    mockRequire(
-      path.join(__dirname, '..', 'package.json'),
-      require(path.join(filesPath, 'package'))
-    );
-  });
+describe('getCommands', () => {
 
   it('list of the commands should be an array', () => {
-    expect(commands).to.be.an('array');
+    expect(getCommands()).to.be.an('array');
   });
 
   it('should contain some pre-defined commands by default', () => {
-    expect(commands).to.be.not.empty;
+    expect(getCommands()).to.be.not.empty;
   });
 
   it('should return a single command (plugin export one command)', () => {
-    mockRequire(testPluginPath, singlePlugin);
+    mock(pluginFolder, commands.single);
     expect(getCommands().length).to.be.equal(1);
-    mockRequire.stop();
   });
 
-  it('should return a single command (plugin export one command)', () => {
-    mockRequire(testPluginPath, multiPlugin);
+  it('should return multiple commands (plugin export an array of commands)', () => {
+    mock(pluginFolder, commands.multiple);
     expect(getCommands().length).to.be.equal(2);
-    mockRequire.stop();
   });
 
-  after(() => {
-    mockFs.restore();
-    mockRequire.stopAll();
+  it('should return an unique list of commands (by name)', () => {
+    mock(
+      path.join(__dirname, '..', 'package.json'),
+      plugins.valid
+    );
+    mock(
+      path.join(process.cwd(), 'node_modules', 'rnpm-plugin-build'),
+      require('../node_modules/rnpm-plugin-link')
+    );
+    expect(getCommands().length).to.be.equal(1);
+  });
+
```

Comment:
I would like to note that `getCommands()` joins `node_module` anyway, so that's the only way to test it. But since it's mocked, that module is not even there (the call is mocked).